File: app/src/main/java/com/sibvic/listernitonce/FileInfoList.java
package com.sibvic.listernitonce;

import java.util.ArrayList;

/**
 * Contains list of FileInfo
 */

class FileInfoList {
    private ArrayList<FileInfo> _listItems = new ArrayList<>();

    int indexOf(FileInfo fileInfo) {
        return _listItems.indexOf(fileInfo);
    }

    void remove(int index) {
        _listItems.remove(index);
    }

    void add(FileInfo fileInfo) {
        _listItems.add(fileInfo);
    }

    ArrayList<FileInfo> toArrayList() {
        return _listItems;
    }

    FileInfo findFileById(String fileId) {
        for (FileInfo file : _listItems) {
            if (file.getId().equals(fileId)) {
                return file;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/sibvic/listernitonce/FileInfoList.kt
package com.sibvic.listernitonce

import java.util.ArrayList

/**
 * Contains list of FileInfo
 */

internal class FileInfoList {
    private val _listItems = ArrayList<FileInfo>()

    fun clear() {
        _listItems.clear()
    }

    fun indexOf(fileInfo: FileInfo): Int {
        return _listItems.indexOf(fileInfo)
    }

    fun remove(index: Int) {
        _listItems.removeAt(index)
    }

    fun add(fileInfo: FileInfo) {
        _listItems.add(fileInfo)
    }

    fun toArrayList(): ArrayList<FileInfo> {
        return _listItems
    }

    fun findFileById(fileId: String): FileInfo? {
        for (file in _listItems) {
            if (file.id == fileId) {
                return file
            }
        }
        return null
    }
}
